src/utils/useOutsideClick.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import noop from 'lodash/noop';
+import { Annotation } from '../@types';
+
+// Returns a hook that checks when a mousedown event is fired on the document whether the mousedown event
+// occurred outside of the provided element reference
+export default function useOutsideClick(
+    annotations: Annotation[],
```

Comment:
Is this dependency needed? We should be able to rely on the root ref changing. It would allow this hook to be reused elsewhere, too.